sdk/eventgrid/azure-eventgrid/azure/eventgrid/_operations/_patch.py

```diff
@@ -187,6 +212,140 @@ def receive_cloud_events(
+        return deserialized # type: ignore
+
+
+def _to_http_request(topic_name: str, **kwargs: Any) -> HttpRequest:   
+    # Create a HTTP request for a binary mode CloudEvent
+
+    _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+    _params = case_insensitive_dict(kwargs.pop("params", {}) or {})
+
+    event = kwargs.pop("event")
+    binary_mode = kwargs.pop("binary_mode", True)
+
+    # Content of the request is the data, if already in binary - no work needed
+    _content = _check_content_type(event.data)
+ 
+    # content_type must be CloudEvent DataContentType when in binary mode
+    default_content_type = kwargs.pop('content_type', _headers.pop('content-type', "application/cloudevents+json; charset=utf-8"))
+    content_type: str = event.datacontenttype or default_content_type
+
+    api_version: str = kwargs.pop('api_version', _params.pop('api-version', "2023-06-01-preview"))
+    accept = _headers.pop('Accept', "application/json")
+
+    # Construct URL
+    _url = "/topics/{topicName}:publish"
+    path_format_arguments = {
+        "topicName": _SERIALIZER.url("topic_name", topic_name, 'str'),
+    }
+
+    _url: str = _url.format(**path_format_arguments)  # type: ignore
+
+    # Construct parameters
+    _params['api-version'] = _SERIALIZER.query("api_version", api_version, 'str')
+
+    # Construct headers
+    _headers['content-type'] = _SERIALIZER.header("content_type", content_type, 'str')
+    _headers['Accept'] = _SERIALIZER.header("accept", accept, 'str')
+    # Cloud Headers
+    _headers['ce-source'] = _SERIALIZER.header('ce-source', event.source, 'str')
+    _headers['ce-type'] = _SERIALIZER.header('ce-type', event.type, 'str')
+    if event.specversion:
+        _headers['ce-specversion'] = _SERIALIZER.header('ce-specversion', event.specversion, 'str')
+    if event.id:
+        _headers['ce-id'] = _SERIALIZER.header('ce-id', event.id, 'str')
+    if event.time:
+        _headers['ce-time'] = _SERIALIZER.header('ce-time', event.time, 'str')
+    if event.dataschema:
+        _headers['ce-dataschema'] = _SERIALIZER.header('ce-dataschema', event.dataschema, 'str')
+    if event.subject:
+        _headers['ce-subject'] = _SERIALIZER.header('ce-subject', event.subject, 'str')
+    if event.extensions:
+        for extension, value in event.extensions.items():
+            _headers[f'ce-{extension}'] = _SERIALIZER.header('ce-extensions', value, 'str')
+
+    return HttpRequest(
+        method="POST",
+        url=_url,
+        params=_params,
+        headers=_headers,
+        content=_content, # pass through content
+        **kwargs
+    )
+
+def _check_content_type(data: Any) -> None:
+    # Check the content type of the data and convert to bytes if needed
+    if isinstance(data, bytes):
+        return data
+    elif isinstance(data, str):
+        return data.encode("utf-8")
+    else:
+        try:
+            return json.dumps(event, cls=AzureJSONEncoder, exclude_readonly=True)  # type: ignore
```

Comment:
we are supposed to be sending binary data, is it enough to json.dumps this, or should I also encode("utf-8") the string after? I know the request sent over the wire as bytes regardless but technically wouldn't this data be a str?

Reply:
`json.dumps` wont give you a utf-8 encoded string (it looks like the encoding kwarg has been removed from the API). To match the behavior from str above, this would need to be encoded too. Is line 347 needed then if its all bytes over the wire anyway?

Reply:
depends on what the HTTPRequest does to content -- will check :)

Reply:
Oh yah why would I need this function if we have this here: https://github.com/Azure/azure-sdk-for-python/blob/68eb2414e77da45653c90852259dd04a5906476b/sdk/core/azure-core/azure/core/rest/_helpers.py#L139 In core